src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -179,17 +179,22 @@ struct DeactivateApplication {
 struct SDLAllowedNotification {
   SDLAllowedNotification(const connection_handler::DeviceHandle& device_id,
                          PolicyManager* policy_manager,
+                         sync_primitives::RecursiveLock& policy_manager_lock,
                          StateController& state_controller)
       : device_id_(device_id)
       , policy_manager_(policy_manager)
+      , policy_manager_lock_(policy_manager_lock)
       , state_controller_(state_controller) {}
 
   void operator()(const ApplicationSharedPtr& app) {
     DCHECK_OR_RETURN_VOID(policy_manager_);
     if (device_id_ == app->device()) {
       std::string hmi_level = "NONE";
       mobile_apis::HMILevel::eType default_mobile_hmi;
-      policy_manager_->GetDefaultHmi(app->policy_app_id(), &hmi_level);
+      {
+        sync_primitives::AutoLock lock(policy_manager_lock_);
```

Comment:
@SNiukalov Add check for shared ptr validity before dereferencing

Reply:
@AByzhynar 
Fixed in: https://github.com/smartdevicelink/sdl_core/pull/2614/commits/a4f0569d8b76a3c5b0abdade1759597790b3e70b